### PR TITLE
[SOAR-9284] IntSights Trigger Fix

### PR DESCRIPTION
**Pull request by cmcnally-r7**
### Description

Describe the proposed changes:

  - Fix `Get Alerts` action not honouring `is_closed` & other boolean tags.
  - Resolve issue on filtering alerts by date.
  - Additional: Fix `get_complete_alert_by_id` returning the wrong value.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [x] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [x] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [x] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [x] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [x] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [x] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [x] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [x] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [x] Work fully completed
- [x] Functional
  - [x] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [x] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `icon-plugin run -A -R all -T all`. For single action validation: `icon-plugin run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`icon-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

**Review by igorski-r7 — CHANGES_REQUESTED** [review]
Looks good! Added a single small comment 😃

File: plugins/rapid7_intsights/icon_rapid7_intsights/util/api.py [hunk 1]
```diff
@@ -38,10 +38,10 @@ def to_dict(self) -> dict:
...
                 "foundDateFrom": int(self.found_date_from) if self.found_date_from else None,
                 "foundDateTo": int(self.found_date_to) if self.found_date_to else None,
-                "assigned": self.assigned == "Assigned" if self.assigned else None,
-                "isFlagged": self.is_flagged == "Flagged" if self.is_flagged else None,
-                "isClosed": self.is_closed == "Closed" if self.is_closed else None,
-                "hasIoc": self.has_ioc,
+                "assigned": json.dumps(self.assigned == "Assigned" if self.assigned else None),
+                "isFlagged": json.dumps(self.is_flagged == "Flagged" if self.is_flagged else None),
+                "isClosed": json.dumps(self.is_closed == "Closed" if self.is_closed else None),
+                "hasIoc": json.dumps(self.has_ioc if self.has_ioc else None),
```
Comment (jkaluza-rapid7) [inline plugins/rapid7_intsights/icon_rapid7_intsights/util/api.py:44]: Please add unit-tests to reflect that change

**Review by jkaluza-rapid7 — CHANGES_REQUESTED** [review]

File: plugins/rapid7_intsights/icon_rapid7_intsights/util/api.py (hunk 1 above)
Comment (cmcnally-r7) [reply plugins/rapid7_intsights/icon_rapid7_intsights/util/api.py:44]: When I ran `coverage run -m unittest *` & `coverage report -m` it seems to be that those lines are already covered by the existing unit tests

File: plugins/rapid7_intsights/icon_rapid7_intsights/util/api.py [hunk 2]
```diff
@@ -98,6 +98,28 @@ def to_dict(self) -> dict:
...
+def subtract_hour(time: int):
+    hour_in_milliseconds = 3600 * 1000
+    minus_hour_in_milliseconds = time - hour_in_milliseconds
+    return minus_hour_in_milliseconds
+
+
+def subtract_week(time: int):
+    week_in_milliseconds = 604800 * 1000
+    minus_week_in_milliseconds = time - week_in_milliseconds
+    return minus_week_in_milliseconds
```
Comment (jkaluza-rapid7) [inline plugins/rapid7_intsights/icon_rapid7_intsights/util/api.py:120]: Looks good, move that to utils

**Review by jkaluza-rapid7 — CHANGES_REQUESTED** [review]

**Review by igorski-r7 — APPROVED** [review]
Looks good! 👍

File: plugins/rapid7_intsights/unit_test/test_new_alert.py [hunk 3]
```diff
@@ -98,3 +108,60 @@ def test_trigger_with_no_inputs(self, make_request, ss):
...
+                "alert_ids": [
+                    "7cafac7ec5adaebf62257a4c",
+                    "7cafac7ec5adaebf62257a4d",
+                    "7cafac7ec5adaebf62257a4e",
+                    "7cafac7ec5adaebf62257a4f",
+                ]
+            }
+        )
+
+        self.action.run({Input.SOURCE_DATE_FROM: "1633047083142"})
```
Comment (igorski-r7) [inline plugins/rapid7_intsights/unit_test/test_new_alert.py:146]: Please connect those both tests together using `parameterized` class above 😃

**Review by igorski-r7 — CHANGES_REQUESTED** [review]

**Review by blaxminarayan-r7 — CHANGES_REQUESTED** [review]

**Review by igorski-r7 — APPROVED** [review]

**Review by jkaluza-rapid7 — APPROVED** [review]

**Pull request merged**